app/web/src/workers/webworker.ts

```diff
@@ -957,6 +942,124 @@ const initIndexAndChangeSet = (
+      ...data,
+      insertedMTM: false,
+      indexExists: false,
+      previousIndexes: "",
+    });
+    // should always be present
+    if (!msg.patch.fromChecksum) {
+      span.end();
+      error("Missing fromChecksum on MvIndex Patch", msg.patch);
+      return;
+    }
+
+    // if we don't *already* have the toIndexChecksum stored
+    // that means we never got the MV data we care about for this index
+    // so the patch is useless to us, abort
+    const indexQuery = db.exec({
+      sql: `select checksum from indexes where checksum = ?`,
+      returnValue: "resultRows",
+      bind: [msg.meta.toIndexChecksum],
+    });
+    const indexExists = oneInOne(indexQuery);
+    if (indexExists) span.setAttribute("indexExists", indexExists?.toString());
+    else {
+      span.end();
+      debug(
+        `${msg.meta.toIndexChecksum} doesn't exist, ignoring index patch WTF`,
+      );
+      return;
+    }
+
+    // if we don't have the fromChecksum, we can't patch
+    // Ragnarok the world has ended, fetch the full index
+    const previousIndexes = workspaceAtomExistsOnIndexes(
+      db,
+      msg.patch.kind,
+      msg.meta.workspaceId,
+      msg.patch.fromChecksum,
+    );
+    span.setAttribute("previousIndexes", JSON.stringify(previousIndexes));
+    if (previousIndexes.length === 0) {
+      debug(
+        `Cannot patch, atom not found at ${msg.patch.fromChecksum}: ${previousIndexes}`,
+      );
+      await niflheim(db, msg.meta.workspaceId, msg.meta.changeSetId);
+      span.end();
+      return;
+    }
+
+    // PASSED CHECKS, patch it
+    const atom: Required<WorkspaceAtom> = {
+      ...data,
+      fromChecksum: msg.patch.fromChecksum, // duplicated for tsc
+      operations: msg.patch.patch,
+      id: msg.meta.workspaceId,
+    };
+    patchAtom(db, atom);
+    const inserted = insertAtomMTM(db, atom, msg.meta.toIndexChecksum);
+    span.setAttribute("insertedMTM", inserted);
+    span.end();
+
+    // don't move any indexes on the changeset record, thats taken care of elsewhere
+
+    const patchedIndex = get(
+      db,
+      msg.meta.workspaceId,
+      msg.meta.changeSetId,
+      EntityKind.MvIndex,
+      msg.meta.workspaceId,
+    );
+    if (patchedIndex === -1) {
+      error("Index not found after writing", msg.meta.workspaceId);
+      span.setAttribute("readFailure", true);
+      span.end();
+      return;
+    }
+
+    // now delete any atom mtms that dont exist in the index
+    const placeholders: string[] = [];
+    const bind: string[] = [];
+    (patchedIndex as StoredMvIndex).mvList.forEach((atom) => {
+      placeholders.push("(?, ?, ?, ?)");
+      bind.push(msg.meta.toIndexChecksum, atom.kind, atom.id, atom.checksum);
+    });
+    const sql = `
+      delete from index_mtm_atoms
+      where 
+        (
+          index_checksum,
+          kind,
+          args,
+          checksum,
+        )
+      NOT IN
+        (
+          (${placeholders.join("), (")})
+        );
+    `;
```

Comment:
Won't this delete all the index atom mtms that we have in place for *other* change sets than the one that we just got an index patch for?